src/Ledger/Conway/Specification/RewardUpdate/Properties.lagda.md

```diff
@@ -0,0 +1,39 @@
+```agda
+{-# OPTIONS --safe #-}
+
+open import Ledger.Conway.Specification.Transaction using (TransactionStructure)
+open import Ledger.Conway.Specification.Abstract using (AbstractFunctions)
+
+module Ledger.Conway.Specification.RewardUpdate.Properties
+  (txs : _) (open TransactionStructure txs)
+  (abs : AbstractFunctions txs) (open AbstractFunctions abs)
+  where
+open import Ledger.Conway.Specification.RewardUpdate txs abs
+open import Ledger.Conway.Specification.Epoch txs abs
+open import Ledger.Conway.Specification.Epoch.Properties txs abs
+open import Ledger.Prelude
+
+open Computational ⦃...⦄
+
+instance
+  Computational-RUPD : Computational _⊢_⇀⦇_,RUPD⦈_ String
+  Computational-RUPD .computeProof _ (just ru) _ =
+    success (just ru , RUPD-Reward-Update-Exists)
+  Computational-RUPD .computeProof _ nothing slot
+    with firstSlot (epoch slot) + StabilityWindow <? slot
+  ... | yes p  =
+    let ru' = _
+     in success (just ru' , RUPD-Create-Reward-Update {ru' = ru'} p)
+  ... | no ¬p = success (nothing , RUPD-Reward-Too-Early {! (≮⇒≥ {A = Slot} _ ¬p) !})
```

Comment:
Needs `HasTotalOrder` for Slot. Perhaps this can be avoided if we redefine `RUPD-Reward-Too-Early` to use `≮` instead of `≥`.